script/LaunchProjectAndDeployDomJuice.s.sol

```diff
@@ -0,0 +1,145 @@
+import {IJBFundingCycleBallot} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBFundingCycleBallot.sol";
+// import {IJBOperatorStore} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBOperatorStore.sol";
+import {IJBProjects} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBProjects.sol";
+import {IJBSingleTokenPaymentTerminalStore3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBSingleTokenPaymentTerminalStore3_1_1.sol";
+import {IJBPaymentTerminal} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBPaymentTerminal.sol";
+import {JBETHPaymentTerminal3_1_1} from "@jbx-protocol/juice-contracts-v3/contracts/JBETHPaymentTerminal3_1_1.sol";
+import {JBFundAccessConstraints} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundAccessConstraints.sol";
+import {JBFundingCycleData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundingCycleData.sol";
+import {JBFundingCycleMetadata} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundingCycleMetadata.sol";
+import {JBGlobalFundingCycleMetadata} from
+    "@jbx-protocol/juice-contracts-v3/contracts/structs/JBGlobalFundingCycleMetadata.sol";
+import {JBGroupedSplits} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBGroupedSplits.sol";
+//import {JBOperatable} from "@jbx-protocol/juice-contracts-v3/contracts/abstract/JBOperatable.sol";
+//import {JBOperations} from "@jbx-protocol/juice-contracts-v3/contracts/libraries/JBOperations.sol";
+//import {JBOperatorData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBOperatorData.sol";
+import {JBProjectMetadata} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBProjectMetadata.sol";
+//import {LaunchProjectData} from "./structs/LaunchProjectData.sol";
+
+contract LaunchAndDeploy is Script {
+    // Juicebox contracts
+    struct Contracts {
+        IJBController3_1 controller; // Controller that configures funding cycles
+        IJBSingleTokenPaymentTerminalStore3_1_1 paymentTerminalStore; // Stores all payment terminals
+        JBETHPaymentTerminal3_1_1 ethPaymentTerminal; // Default ETH payment terminal
+    }
+
+    Contracts contracts;
+
+    // JB Project Launch struct array parameters
+    IJBPaymentTerminal[] _terminals; // default empty
+    JBGroupedSplits[] _groupedSplits; // Default empty
+    JBFundAccessConstraints[] _fundAccessConstraints; // Default empty
+
+    // Campaign parameters
+    address payable owner = payable(msg.sender);
+    uint256 cycleTarget = 1200 ether;
+    uint256 startTime = block.timestamp + 172800; // 48 hours from script execution
+    uint256 duration = 1209600; // Two weeks of seconds
+    uint256 minimumPledgeAmount = 50 ether;
+
+    event DelegateDeployed(uint256 _projectID, address _delegate, address _owner);
+
+    function run() external returns (uint256 projectID) {
```

Comment:
do we really need this function that uses hard-coded values vs. just calling launchProjectFor directly from the command-line?

Reply:
Do you mean directly calling the Juicebox controller.launchProjectFor() function from the CLI?... Or, if you mean calling the script’s function from the CLI, please educate me, cuz I didn’t know you could do that. That would be very cool. I thought you could only run them just as a single unit, no arguments from CLI?

Reply:
I have no problem with the `run` function. I have a problem with a script hard-coding values since it means you can't do anything different for eg different testnet implementations. As I mention below, if `run` is the entry point, I would expect the script runner to support some sort of arguments to be passed in

Reply:
Figured out how to pass in args to the run function! Man, had to dig in the Forge Github PRs to find the info. They haven't added it to the manual yet. But yeah, I'll be able to make all these parameters programmable and not hardcoded. Yayyy.